feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -311,6 +313,95 @@ static struct vif_crypto {
 	{ "wpa3-mixed", OVSDB_SECURITY_ENCRYPTION_WPA3_EAP, OVSDB_SECURITY_MODE_MIXED, 1 },
 };
 
+extern ovsdb_table_t table_APC_State;
+extern json_t* ovsdb_table_where(ovsdb_table_t *table, void *record);
+
+static int vif_config_proxy_security_set(struct blob_buf *b,
+				    const struct schema_Wifi_VIF_Config *vconf)
+{
+	const char *encryption = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_ENCRYPT);
+	const char *mode = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_MODE);
+	unsigned int i;
+	unsigned int acct_interval;
+	const char *auth_server, *auth_port, *auth_secret, *security_key, *acct_server;
+	struct schema_APC_State apc_conf;
+	const char *local_server = "127.0.0.1";
+
+    	json_t *where = ovsdb_table_where(&table_APC_State, &apc_conf);
+	if (false == ovsdb_table_select_one_where(&table_APC_State,
+			where, &apc_conf)) {
+		LOG(INFO, "APC_State read failed");
+		return -1;
+	}
+	if (!strncmp(apc_conf.mode, "DR", 2)) {
+		auth_server = local_server;
+		acct_server = local_server;
+	} else if (!strncmp(apc_conf.mode, "SR", 2) ||
+		   !strncmp(apc_conf.mode, "BDR", 2)) {
+		auth_server = apc_conf.dr_addr;
+		acct_server = apc_conf.dr_addr;
+	}
+	else {
+		auth_server = local_server;
+		acct_server = local_server;
+	}
+
+	if (!strcmp(encryption, OVSDB_SECURITY_ENCRYPTION_OPEN) || !mode)
```

Comment:
This can be re-organized so that all the security setting code is not duplicated.  The existing vif_config_security_set() could check if RadSec proxy is enabled and then adjust the addresses accordingly.

Duplicating this code will be a maintenance problem as new security methods are added.

Reply:
Done.

Reply:
Refactored and removed duplicate code.